observability/alerting.md

```diff
@@ -0,0 +1,14 @@
+# Alerting and Observability
+
+One of the goals of building a highly observable systems is to provide valuable insight into the behavior of the application. The need for observable systems is to identify problem areas and surface them by raising appropriate alerts before it starts impacting end users.
+
+## Best Practices
+
+- The foremost thing to do before creating alerts is to implement observability. Without monitoring systems in place, it becomes next to impossible to know what activities needs to be monitored and when to alert the teams.
+- Identify what the application's minimum viable service quality needs to be. It is not what you intend to deliver, but is acceptable for the customer. These [Service Level Objectives](https://landing.google.com/sre/sre-book/chapters/service-level-objectives/)(SLOs) is a metric for measurement of the application's performance.
+- SLOs are defined with respect to the end users. The alerts must watch for visible impact to the user. For example, alerting on request rate, latency and errors.
+- Use automated, scriptable tools mimic end-to-end important code paths relatable to activities in the application. Create alert polices on user impacting events or metric rate of change.
+- Alert fatigue is real. Engineers are recommended to pay attention to their monitoring system so that right alerts and thresholds can be defined.
```

Comment:
```suggestion
- Alert fatigue is real. Engineers are recommended to pay attention to their monitoring system so that accurate alerts and thresholds can be defined.
```